### PR TITLE
update devDependencies

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo

Only thing I couldn't update was cucumber as then cucumber_lint would be using a conflicting version of gherkin. I'll work on updating cucumber_lint.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,14 +3,14 @@ source 'http://rubygems.org'
...
-  gem 'cucumber'
-  gem 'cucumber_lint'
-  gem 'kappamaki'
-  gem 'mortadella'
-  gem 'open4'
-  gem 'parallel_tests'
-  gem 'rake'
-  gem 'rubocop'
-  gem 'rspec'
+  gem 'activesupport', '4.2.6'
```
Comment (kevgo) [inline Gemfile]: In my opinion this is an anti-pattern in Ruby. The versions are locked down in Gemfile.lock. The Gemfile should only lock down versions if there is a good reason for staying on that version, even if other dependencies get updated. Like "version X of this gem is broken, lets avoid it".

Comment (charlierudolph) [inline Gemfile]: The main reason I did this was it was actually able to control which versions of which get installed via `bundle install`. When I ran bundle update, it updated cucumber and rolled back cucumber_lint in order for the two to use a compatible gherkin. It doesn't necessarily need to be locked down here but following what we did with node, why not also lock down here. That way you don't need to look at the gemfile.lock to see what versions of what you are using.

Comment (kevgo) [inline Gemfile]: In my opinion Bundler has solved this the best way, and Node is an inferior and broken solution in regards to versioning. Usually the particular versions of gems don't matter much, except when updating. And then you can see the changes in the diff of Gemfile.lock. In this case we want to hard-code the version of Cucumber, so let's do that here, and leave the other gems open. Let's stay idiomatic here.

**Comment by charlierudolph** [comment]
@kevgo updated

File: features/support/env.rb [hunk 2]
```diff
@@ -6,25 +6,16 @@
...
 
-DEBUG = {
-
-  # Prints debug info for all activities
-  all: ENV['DEBUG'],
-
-  # Prints debug info only for the Git commands run
-  commands_only: ENV['DEBUG_COMMANDS']
-}
-
```
Comment (charlierudolph) [inline features/support/env.rb]: Rubocop wanted this frozen since its supposedly a constant. Changed how it works.

File: Gemfile [hunk 3]
```diff
@@ -4,13 +4,13 @@ ruby '2.2.3'
 
 group :development do
   gem 'activesupport'
-  gem 'cucumber'
-  gem 'cucumber_lint'
+  gem 'cucumber', '2.1.0'
```
Comment (kevgo) [inline Gemfile]: Is it possible to use semantic versioning here? `~> 2.1.0`

File: features/support/env.rb (hunk 2 above)
Comment (kevgo) [inline features/support/env.rb]: Yeah, all uppercase in Ruby means constant.

File: features/support/env.rb [hunk 4]
```diff
@@ -68,6 +59,18 @@ def initialize_environment
 Before do
   @error_expected = false
   @non_empty_stash_expected = false
+  @debug = ENV['DEBUG']
+  @debug_commands = ENV['DEBUG_COMMANDS']
+end
+
+
+Before '@debug' do
```
Comment (kevgo) [inline features/support/env.rb]: This is much better anyways!

**Comment by kevgo** [comment]
Cool!  :ship:

File: Gemfile (hunk 3 above)
Comment (charlierudolph) [inline Gemfile]: We can probably use 2.2.0 but its 2.3.0 that starts using the incompatible gherkin

Comment (charlierudolph) [inline Gemfile]: 2.1.0 was just what we were using previously

Comment (kevgo) [inline Gemfile]: `< 2.3.0` ?

Comment (kevgo) [inline Gemfile]: Or `~> 2.2.0`

**Pull request merged**